pkg/registry/core/service/storage/rest_test.go

```diff
@@ -1839,173 +1257,112 @@ func TestServiceRegistryList(t *testing.T) {
+	_, _, err = storage.Delete(ctx, createdSvc1.Name, rest.ValidateAllObjectFunc, &metav1.DeleteOptions{})
 	if err != nil {
 		t.Errorf("Unexpected error deleting service: %v", err)
 	}
 
-	svc2 := &api.Service{
-		ObjectMeta: metav1.ObjectMeta{Name: "bar"},
-		Spec: api.ServiceSpec{
-			Selector:        map[string]string{"bar": "baz"},
-			SessionAffinity: api.ServiceAffinityNone,
-			Type:            api.ServiceTypeClusterIP,
-			Ports: []api.ServicePort{{
-				Port:       6502,
-				Protocol:   api.ProtocolTCP,
-				TargetPort: intstr.FromInt(6502),
-			}},
-		},
-	}
+	svc2 := svctest.MakeService("bar", svctest.SetClusterIPs(svc1.Spec.ClusterIP))
 	ctx = genericapirequest.NewDefaultContext()
-	createdSvc2, _ := storage.Create(ctx, svc2, rest.ValidateAllObjectFunc, &metav1.CreateOptions{})
-	createdService2 := createdSvc2.(*api.Service)
-	if createdService2.Name != "bar" {
-		t.Errorf("Expected bar, but got %v", createdService2.Name)
+	obj, err = storage.Create(ctx, svc2, rest.ValidateAllObjectFunc, &metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("error creating service: %v", err)
+	}
+	createdSvc2 := obj.(*api.Service)
+	if createdSvc2.Name != "bar" {
+		t.Errorf("Expected bar, but got %v", createdSvc2.Name)
 	}
-	if !makeIPNet(t).Contains(net.ParseIP(createdService2.Spec.ClusterIPs[0])) {
-		t.Errorf("Unexpected ClusterIP: %s", createdService2.Spec.ClusterIPs[0])
+	if !makeIPNet(t).Contains(net.ParseIP(createdSvc2.Spec.ClusterIPs[0])) {
+		t.Errorf("Unexpected ClusterIP: %s", createdSvc2.Spec.ClusterIPs[0])
 	}
 }
 
 func TestServiceRegistryIPUpdate(t *testing.T) {
-	storage, _, server := NewTestREST(t, nil, singleStackIPv4)
+	storage, server := NewTestREST(t, singleStackIPv4)
 	defer server.Terminate(t)
 
-	svc := &api.Service{
-		ObjectMeta: metav1.ObjectMeta{Name: "foo", ResourceVersion: "1"},
-		Spec: api.ServiceSpec{
-			Selector:        map[string]string{"bar": "baz"},
-			SessionAffinity: api.ServiceAffinityNone,
-			Type:            api.ServiceTypeClusterIP,
-			Ports: []api.ServicePort{{
-				Port:       6502,
-				Protocol:   api.ProtocolTCP,
-				TargetPort: intstr.FromInt(6502),
-			}},
-		},
-	}
+	svc := svctest.MakeService("foo")
 	ctx := genericapirequest.NewDefaultContext()
-	createdSvc, _ := storage.Create(ctx, svc, rest.ValidateAllObjectFunc, &metav1.CreateOptions{})
+	createdSvc, err := storage.Create(ctx, svc, rest.ValidateAllObjectFunc, &metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
 	createdService := createdSvc.(*api.Service)
-	if createdService.Spec.Ports[0].Port != 6502 {
-		t.Errorf("Expected port 6502, but got %v", createdService.Spec.Ports[0].Port)
+	if createdService.Spec.Ports[0].Port != svc.Spec.Ports[0].Port {
+		t.Errorf("Expected port %d, but got %v", svc.Spec.Ports[0].Port, createdService.Spec.Ports[0].Port)
 	}
 	if !makeIPNet(t).Contains(net.ParseIP(createdService.Spec.ClusterIPs[0])) {
```

Comment:
this is covered in alloc. we should remove.

Reply:
defer